homeassistant/components/zwave_js/config_flow.py

```diff
@@ -62,14 +110,268 @@ async def async_step_user(
+            else:
+                # Sleep some seconds to let the add-on start properly before connecting.
+                await asyncio.sleep(ADDON_SETUP_TIME)
+                discovery_info = await self._async_get_addon_discovery_info()
+                self.ws_address = (
+                    f"ws://{discovery_info['host']}:{discovery_info['port']}"
+                )
+
+                if not self.unique_id:
+                    try:
+                        version_info = await async_get_version_info(
+                            self.hass, self.ws_address
+                        )
+                    except CannotConnect:
+                        return self.async_abort(reason="cannot_connect")
+                    await self.async_set_unique_id(
+                        version_info.home_id, raise_on_progress=False
+                    )
+
+                self._abort_if_unique_id_configured()
+                return self._async_create_entry_from_vars()
+
+        usb_path = self.addon_config.get(CONF_ADDON_DEVICE, self.usb_path or "")
+        network_key = self.addon_config.get(
+            CONF_ADDON_NETWORK_KEY, self.network_key or ""
+        )
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USB_PATH, default=usb_path): str,
+                vol.Optional(CONF_NETWORK_KEY, default=network_key): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="start_addon", data_schema=data_schema, errors=errors
+        )
+
+    async def _async_get_addon_info(self) -> dict:
+        """Return and cache Z-Wave JS add-on info."""
+        assert self.hass
+        try:
+            addon_info: dict = await self.hass.components.hassio.async_get_addon_info(
+                "core_zwave_js"
+            )
+        except self.hass.components.hassio.HassioAPIError as err:
+            _LOGGER.error("Failed to get Z-Wave JS add-on info: %s", err)
+            raise AbortFlow("addon_info_failed") from err
+
+        return addon_info
+
+    async def _async_is_addon_running(self) -> bool:
+        """Return True if Z-Wave JS add-on is running."""
+        addon_info = await self._async_get_addon_info()
+        return bool(addon_info["state"] == "started")
+
+    async def _async_is_addon_installed(self) -> bool:
+        """Return True if Z-Wave JS add-on is installed."""
+        addon_info = await self._async_get_addon_info()
+        return addon_info["version"] is not None
+
+    async def _async_get_addon_config(self) -> dict:
+        """Get Z-Wave JS add-on config."""
+        addon_info = await self._async_get_addon_info()
+        return cast(dict, addon_info["options"])
+
+    async def _async_set_addon_config(self, config: dict) -> None:
+        """Set Z-Wave JS add-on config."""
+        assert self.hass
+        options = {"options": config}
+        try:
+            await self.hass.components.hassio.async_set_addon_options(
```

Comment:
We should prefer to import the `hassio` component and avoid using `hass.components.X`. The latter can't be typed and so typechecking would miss it if we change method signatures or remove them.

Reply:
Downside is that new helpers can't be tested in custom integrations without this. That's how I test on a real Supervisor host. This helper is already released so it's ok here.

Reply:
Why wouldn't that work? `hass.components.hassio` just does a `from hass.components import hassio` under the hood and wraps methods decorated with `@bind_hass`.

Reply:
If we add a new method in the hassio integration and copy that as a custom integration, the zwave_js integration will look in the wrong directory if we're using normal top level imports. I got hit by this when developing the hassio add-on helpers for ozw.

Reply:
Ah yes, that is correct. You can also connect your dev machine to a supervisor on another machine: https://developers.home-assistant.io/docs/supervisor/development#integration-development